### ISSUE TITLE
what Auto-Weka reports exactly?

### ISSUE THREAD

**Issue by karruo** [larskotthoff/autoweka#36]
I am looking the AutoWeka package that looks awesome. I tested it for a few datasets of a regression problem and then became uncertain how to interpret the results reported. Specifically, I am running the package in WEKA classifier GUI (the general classifier tab, not the special AutoWeka). I used 10 parallel runs in one test and let it run for 24 hours and then at the end AutoWeka gave me a report with expected mean absolute error and root mean square error etc. But to what this report refers to? I have learned AutoWeka splits the data to training and test sets and then further uses 10-fold CV in the training set to optimise the hyperparameters. I assume then the test set is used to check the performance further. So, my question is if the results reported in the WEKA GUI results buffer are 1) results based on the test set evaluation within each run?, 2) results from the individual CV folds of the training part or 3) full trained model to entire data? It seems difficult to find this information from the documentation. I have learned that AutoWeka saves more detailed information to the /tmp folder but I have not found any documentation for the files there either - one needs to guess what they could mean. E.g. there is a file called predictions.csv which I guess are the CV fold predictions on the training set - but then are there somewhere the test set predictions available as well?

I am hesitating in using this magnificant package as I a m not sure what the results on the prediction capability mean. I would appreciate some help or pointers to further reading if anyone can provide.

regards, Kari

**Comment by larskotthoff** [comment]
The reported performance is on an internal held out test set (1/3 of the entire data).

**Comment by karruo** [comment]
Thanks for a quick and clear reply. I have a follow-up question to further clarify the output. When I then choose "Visualize classifier errors" by right-clicking the results list item I get a graph with the number of cases matching the number of original observations, not one third of observations. I assume then that this graph does not correspond to the test case evaluation as there are many more cases? If not what the "error garph" cases represent - a trained model on the full data set, perhaps? I wonder if the test set predictions (n/3) are available somewhere?

regards, Kari

**Comment by larskotthoff** [comment]
This is a bit weird unfortunately because WEKA does its own evaluation. The number that Auto-WEKA reports is its internal performance estimate, and in addition to that you get WEKA's performance estimate based on the returned model (which is determined by the identified best classifier and parameters). How exactly this is computed depends on the evaluation options you select in WEKA, and Auto-WEKA has no influence over this. We currently provide no (easy) way to get the predictions from Auto-WEKA's internal evaluations.
